### PR TITLE
Wqp 1643 add summary service to station pages

### PR DESCRIPTION
**Pull request by abriggs-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

WQP-1643 Add Summary Service Table to Station Pages
-----------
Adds a table to the station pages that show the data collection period of record by characteristic type for the location

Description
-----------
Uses the Pandas data library to open and sort the records of the comma separated file (CSV) from Water Quality Portal Summary Service

Image - sample table
image

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: server/wqp/portal_ui_blueprint/templates/site.html [hunk 1]
```diff
@@ -69,13 +69,40 @@ <h1>{{ site['MonitoringLocationName'] }} ({{ site['MonitoringLocationIdentifier'
 
         To download the metadata about this site along with water quality data, go to the <a href="{{ url_for('portal_ui.home') }}">Portal Page</a> and enter "{{ site['MonitoringLocationIdentifier'] }}" into the "Site ID" box under Site Parameters
 
-        <h2>Upstream and downstream stations</h2>
+        <h2  class="section-heading">What other monitoring locations are upstream or downstream from this one?</h2>
+        <div class="subheading">Upstream and downstream locations</div>
```
Comment (mbucknell) [inline server/wqp/portal_ui_blueprint/templates/site.html:73]: Do you think this should be a h3 tag?

File: server/wqp/portal_ui_blueprint/templates/site.html [hunk 2]
```diff
@@ -69,13 +69,40 @@ <h1>{{ site['MonitoringLocationName'] }} ({{ site['MonitoringLocationIdentifier'
...
+                        </tr>
+                    {% endfor %}
+                </tbody>
+            </table>
+            <span class="characteristic-group-table-footnote"><sup>&ast;</sup>Date ranges may contain gap years</span>
+        </div>
 
-        <h2>Detailed Metadata</h2>
+        <h2 class="section-heading">What other information is available for this location?</h2>
+        <div class="subheading">Detailed Metadata</div>
```
Comment (mbucknell) [inline server/wqp/portal_ui_blueprint/templates/site.html:105]: Same question as above about h3 tag.

File: server/wqp/utils.py [hunk 3]
```diff
@@ -378,3 +378,49 @@ def delete_old_files(files):
...
+    """
+    Function does the following -
+    1) The data call returns mostly unneeded data, so starts by keeping only the needed columns.
+    2) Adds new columns for the start and end years of the period of record by grabbing the minimum and maximum values
+        for each characteristic group from the YearSummarized. Note: Every 'year summarized' has its own row in the
+        data at this point. The 'startYear' and 'endYear' values will be the same for every row in within a group
+        of 'characteristicType' rows of the same type.
+    3) Uses the min function to keep only one row in each characteristic group.
+    4) Cleans up by dropping the unneeded 'YearSummarized' column.
+    :param period_of_record_summary_data:
```
Comment (mbucknell) [inline server/wqp/utils.py:402]: period_of_record_summary_data is a pandas DataFrame?

**Review by mbucknell — APPROVED** [review]
Consider using h3 tag for subheadings, otherwise it looks good to go!

File: server/wqp/utils.py (hunk 3 above)
Comment (abriggs-usgs) [reply server/wqp/utils.py:402]: Yes. I'll add that.

File: server/wqp/portal_ui_blueprint/templates/site.html (hunk 1 above)
Comment (abriggs-usgs) [reply server/wqp/portal_ui_blueprint/templates/site.html:73]: Well, I actually had the h3 first, then I questioned myself so I read an article on semantic tagging and it was like 'well, some people do it this way, but the 'h' tags are headers for sections and not sub titles. So then I back tracked from the h3. I really don't know what is correct. Happy to go with the 'h3' if you think that is consistent with the pattern we normally follow.

Comment (mbucknell) [reply server/wqp/portal_ui_blueprint/templates/site.html:73]: I see what you are saying. It's not really a sub heading as much as an explanation. Sounds like you thought about it so I'm fine with leaving as is.

**Pull request merged**